apps/angular/src/app/features/anime/anime-dashboard/anime-dashboard.component.ts

```diff
@@ -22,10 +34,97 @@ export class AnimeDashboardComponent {
 		'status',
 	];
 
+	/** Initial anime sorting. */
+	protected readonly initialSortingParameters: SortingParameters<AnimeSortingField>;
+
+	/** Anime filters form group. */
+	protected readonly animeFiltersFormGroup: FormGroup<{
+		search: FormControl<string | null>;
+		animeTypes: FormControl<AnimeType[] | null>;
+	}>;
+
+	/** Anime types. */
+	protected readonly animeTypes: readonly AnimeType[] = EnumUtils.toArray(AnimeType);
+
 	/** Observable for anime previews. */
-	public readonly animePreviews$: Observable<Anime[]>;
+	protected readonly paginatedAnime$: Observable<Pagination<Anime>>;
+
+	/** Paginator settings. */
+	protected readonly paginatorSettings$: Observable<PaginationParameters>;
+
+	public constructor(
+		protected readonly animeParametersService: AnimeParametersService,
+		private readonly destroyRef: DestroyRef,
+		animeService: AnimeService,
+	) {
+		const initialAnimeParameters = this.animeParametersService.animeParameters;
+
+		this.initialSortingParameters = initialAnimeParameters;
+
+		this.animeFiltersFormGroup = new FormGroup({
+			search: new FormControl(initialAnimeParameters.search, { updateOn: 'blur' }),
+			animeTypes: new FormControl<AnimeType[]>([...initialAnimeParameters.animeTypes]),
+		});
+
+		this.paginatedAnime$ = this.animeParametersService.animeParameters$.pipe(
+			switchMap(parameters => animeService.getAnimeList(parameters)),
+		);
+
+		this.paginatorSettings$ = this.animeParametersService.animeParameters$;
+	}
+
+	/** @inheritdoc */
+	public ngOnInit(): void {
+		this.subscribeToFiltersChanges();
+	}
+
+	/**
+	 * Handle pagination parameters change event.
+	 * @param paginationEvent - Pagination event.
+	 */
+	protected handlePaginationParametersChange(paginationEvent: PageEvent): void {
+		const pageNumber: number = paginationEvent.pageIndex + 1;
```

Comment:
```suggestion
		const pageNumber = paginationEvent.pageIndex + 1;
```

In such cases you don't need to describe the type, TS will determine it itself